cow_py/contracts/order.py

```diff
@@ -0,0 +1,243 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Literal, Optional, Union
+
+from eth_account.messages import _hash_eip191_message, encode_typed_data
+from eth_typing import Hash32, HexStr
+from eth_utils.conversions import to_bytes, to_hex
+
+
+@dataclass
+class Order:
+    # Sell token address.
+    sellToken: str
+    # Buy token address.
+    buyToken: str
+    # An optional address to receive the proceeds of the trade instead of the
+    # owner (i.e. the order signer).
+    receiver: str
+    # The order sell amount.
+    #
+    # For fill or kill sell orders, this amount represents the exact sell amount
+    # that will be executed in the trade. For fill or kill buy orders, this
+    # amount represents the maximum sell amount that can be executed. For partial
+    # fill orders, this represents a component of the limit price fraction.
+    #
+    sellAmount: int
+    # The order buy amount.
+    #
+    # For fill or kill sell orders, this amount represents the minimum buy amount
+    # that can be executed in the trade. For fill or kill buy orders, this amount
+    # represents the exact buy amount that will be executed. For partial fill
+    # orders, this represents a component of the limit price fraction.
+    #
+    buyAmount: int
+    # The timestamp this order is valid until
+    validTo: int
+    # Arbitrary application specific data that can be added to an order. This can
+    # also be used to ensure uniqueness between two orders with otherwise the
+    # exact same parameters.
+    appData: str
+    # Fee to give to the protocol.
+    feeAmount: int
+    # The order kind.
+    kind: str
+    # Specifies whether or not the order is partially fillable.
+    partiallyFillable: bool = False
+    # Specifies how the sell token balance will be withdrawn. It can either be
+    # taken using ERC20 token allowances made directly to the Vault relayer
+    # (default) or using Balancer Vault internal or external balances.
+    sellTokenBalance: Optional[str] = None
+    # Specifies how the buy token balance will be paid. It can either be paid
+    # directly in ERC20 tokens (default) in Balancer Vault internal balances.
+    buyTokenBalance: Optional[str] = None
+
+
+# Gnosis Protocol v2 order cancellation data.
+@dataclass
+class OrderCancellations:
+    orderUids: bytearray
+
+
+# Marker address to indicate that an order is buying Ether.
+#
+# Note that this address is only has special meaning in the `buyToken` and will
+# be treated as a ERC20 token address in the `sellToken` position, causing the
+# settlement to revert.
+BUY_ETH_ADDRESS = "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE"
```

Comment:
Is this needed? I tried using this in the API and it always throws errors.

Reply:
The current interface of the cow swap seems to use this approach:

<img width="367" alt="image" src="https://github.com/user-attachments/assets/2e7ff4f3-5ab4-4561-912f-aa52e79d7035">

Reply:
Can you write a simple integration test for this? I recall this address didn't work for swapping xDAI on CoW.

Reply:
Did you try to sell or buy xDAI? This address can just be used to buy. Actually, this address isn't used on any package function, so I will just remove it for now.